### DISCUSSION TITLE
Releases

### DISCUSSION THREAD

**Discussion by kobylynskyi** [kobylynskyi/graphql-java-codegen#430] [🚀 Releases]
Discussions about previous releases (breaking changes, migration, etc) and future releases (scope, plans, release dates, etc)

**Comment by kobylynskyi** [comment]
# :rocket: [4.0.1](https://github.com/kobylynskyi/graphql-java-codegen/releases/tag/v4.0.1)

**Reply by kobylynskyi** [reply]
@jxnu-liguobin, I've released the library + maven and gradle plugins, feel free to release sbt plugin :shipit:

**Comment by kobylynskyi** [comment]
# :rocket: [4.1.0](https://github.com/kobylynskyi/graphql-java-codegen/releases/tag/v4.1.0)

**Reply by kobylynskyi** [reply]
@jxnu-liguobin, library, maven and gradle plugins were released. Please proceed with sbt plugin release :shipit:

**Reply by jxnu-liguobin** [reply]
Is there a problem with SBT publishing? The other two plugins I see have been successful. @kobylynskyi

**Reply by kobylynskyi** [reply]
Please publish SBT 4.1.4 manually this time. I will work on it later if you don’t mind

**Reply by jxnu-liguobin** [reply]
I'm not in a hurry

**Comment by kobylynskyi** [comment]
# :rocket: [4.1.5](https://github.com/kobylynskyi/graphql-java-codegen/releases/tag/v4.1.5)

**Reply by kobylynskyi** [reply]
@jxnu-liguobin do you plan to introduce any new fixes/changes this days? Because I am planning to release [4.1.5](https://github.com/kobylynskyi/graphql-java-codegen/milestone/32) today or tomorrow.

**Reply by jxnu-liguobin** [reply]
@kobylynskyi don't have more.

**Reply by jxnu-liguobin** [reply]
@kobylynskyi By the way, when I was preparing to solve this [issue](https://github.com/kobylynskyi/graphql-java-codegen/issues/582), I thought of a problem. If I need to generate code such as java9, but because the API is not in java8, how can I solve it? Of course, What I'm facing now is 2.12 - > 2.13 in scala, so I can't modify the template directly, because it will cause people who use 2.12 to be unable to use the library. I'm not going to upgrade at the moment, although it will trigger a lot of warnings

**Reply by kobylynskyi** [reply]
@jxnu-liguobin I've released 4.1.5 library and maven/gradle plugins. Please proceed with releasing sbt plugin. Thanks.

**Reply by jxnu-liguobin** [reply]
ok, I will do it, and  when will it be released automatically?

**Reply by kobylynskyi** [reply]
I have troubles with automatic releases, so I postponed this feature

**Comment by kobylynskyi** [comment]
# :rocket: [4.1.6](https://github.com/kobylynskyi/graphql-java-codegen/releases/tag/v4.1.6)

**Reply by kobylynskyi** [reply]
@jxnu-liguobin please release sbt plugin 4.1.6. Thanks

**Reply by jxnu-liguobin** [reply]
done

**Comment by kobylynskyi** [comment]
# :rocket: [5.0.0](https://github.com/kobylynskyi/graphql-java-codegen/releases/tag/v5.0.0)

**Reply by kobylynskyi** [reply]
@jxnu-liguobin could you please release sbt-plugin for the latest codegen version? Thank you!

**Reply by jxnu-liguobin** [reply]
done

**Reply by jxnu-liguobin** [reply]
@kobylynskyi By the way, can you give me a collaborator? 🦘

**Reply by kobylynskyi** [reply]
@jxnu-liguobin definitely! I gave the collaborator role to you.

**Comment by kobylynskyi** [comment]
# :rocket: [5.1.0](https://github.com/kobylynskyi/graphql-java-codegen/releases/tag/v5.1.0)

**Reply by kobylynskyi** [reply]
@jxnu-liguobin please release SBT plugin for 5.1.0. Thanks!
